bolt-sidecar/src/common/score_cache.rs

```diff
@@ -0,0 +1,228 @@
+use std::{
+    borrow::Borrow,
+    collections::HashMap,
+    fmt::Debug,
+    hash::{BuildHasher, Hash, RandomState},
+    ops::{Deref, DerefMut},
+};
+
+pub struct ScoreCache<
+    const GET_SCORE: isize,
+    const INSERT_SCORE: isize,
+    const UPDATE_SCORE: isize,
+    K,
+    V,
+    S = RandomState,
+> {
+    map: HashMap<K, (V, isize), S>,
+    max_len: usize,
+}
+
+// -------- TRAITS --------
+
+impl<const GET_SCORE: isize, const INSERT_SCORE: isize, const UPDATE_SCORE: isize, K, V> Default
+    for ScoreCache<GET_SCORE, INSERT_SCORE, UPDATE_SCORE, K, V, RandomState>
+{
+    fn default() -> Self {
+        ScoreCache::new()
+    }
+}
+
+impl<const GET_SCORE: isize, const INSERT_SCORE: isize, const UPDATE_SCORE: isize, K, V, S> Deref
+    for ScoreCache<GET_SCORE, INSERT_SCORE, UPDATE_SCORE, K, V, S>
+{
+    type Target = HashMap<K, (V, isize), S>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.map
+    }
+}
+
+impl<const GET_SCORE: isize, const INSERT_SCORE: isize, const UPDATE_SCORE: isize, K, V, S> DerefMut
+    for ScoreCache<GET_SCORE, INSERT_SCORE, UPDATE_SCORE, K, V, S>
+{
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.map
+    }
+}
+
+impl<
+        const GET_SCORE: isize,
+        const INSERT_SCORE: isize,
+        const UPDATE_SCORE: isize,
+        K: Debug,
+        V: Debug,
+        S,
+    > Debug for ScoreCache<GET_SCORE, INSERT_SCORE, UPDATE_SCORE, K, V, S>
+{
```

Comment:
is this using the default formatter? looks a bit weird hahah

Reply:
Yes, this is incredibly the default formatter ahah